### PR TITLE
VZ-5677: Add BFS kube-webhook-certgen image

### PR DESCRIPTION
**Pull request by fredtibbitts**
# Description

This image is used to generate webhook certificates for the Prometheus Operator in the event that Cert Manager is disabled. The Prometheus Operator install checks to see if Cert Manager is enabled, and if so sets the helm flag to enable the use of Cert Manager. If Cert Manager is not enabled, this kube-webhook-certgen image will be used to generate the certificates.

While I was in here I also added additional unit test coverage for the promoperator component.

Implements VZ-5677

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [x] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

**Review by michael-cico — APPROVED** [review]

File: platform-operator/controllers/verrazzano/component/prometheus/operator/promoperator_test.go [hunk 1]
```diff
@@ -85,3 +100,84 @@ func TestIsPrometheusOperatorReady(t *testing.T) {
...
+
+	var err error
+	kvs, err = AppendOverrides(ctx, "", "", "", kvs)
+	assert.NoError(t, err)
+	assert.Len(t, kvs, 7)
+
+	assert.Equal(t, "ghcr.io/verrazzano/prometheus-config-reloader", bom.FindKV(kvs, "prometheusOperator.prometheusConfigReloader.image.repository"))
+	assert.NotEmpty(t, bom.FindKV(kvs, "prometheusOperator.prometheusConfigReloader.image.tag"))
+
+	assert.Equal(t, "ghcr.io/verrazzano/alertmanager", bom.FindKV(kvs, "alertmanager.alertmanagerSpec.image.repository"))
```
Comment (desagar) [inline platform-operator/controllers/verrazzano/component/prometheus/operator/promoperator_test.go:137]: is it safe to always assume the ghcr.io/verrazzano prefix in the unit tests?

File: platform-operator/controllers/verrazzano/component/prometheus/operator/promoperator_component.go [hunk 2]
```diff
@@ -41,7 +40,7 @@ func NewComponent() spi.Component {
 			MinVerrazzanoVersion:    constants.VerrazzanoVersion1_3_0,
 			ImagePullSecretKeyname:  "global.imagePullSecrets[0].name",
 			ValuesFile:              filepath.Join(config.GetHelmOverridesDir(), "prometheus-values.yaml"),
-			Dependencies:            []string{certmanager.ComponentName},
```
Comment (davisgomes) [inline platform-operator/controllers/verrazzano/component/prometheus/operator/promoperator_component.go:44]: If Certmanager is enabled, do we need to depend on it for install?

Comment (fredtibbitts) [reply platform-operator/controllers/verrazzano/component/prometheus/operator/promoperator_component.go:44]: No, I don't think we have to wait for Certmanager to be installed. The PO install will add annotations that Certmanager will pick up once it's up and running. At least in my manual testing I haven't seen any issues with the ordering.

File: platform-operator/controllers/verrazzano/component/prometheus/operator/promoperator_test.go (hunk 1 above)
Comment (fredtibbitts) [reply platform-operator/controllers/verrazzano/component/prometheus/operator/promoperator_test.go:137]: The prefix is arbitrary, it just needs to match what's in `test_bom.json`, and the test bom is based on the actual bom vaues.

**Pull request merged**